earth2grid/healpix/nn.py

```diff
@@ -0,0 +1,59 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import math
+
+import einops
+import torch
+
+from earth2grid.healpix._padding import pad
+
+
+def conv2d(input, weight, bias=None, stride=1, padding=0, dilation=1, groups=1):
+    """conv2d(input, weight, bias=None, stride=1, padding=0, dilation=1, groups=1) -> Tensor
+
+    Applies a 2D convolution over an input image composed of several input
+    planes.
+
+    This operator supports :ref:`TensorFloat32<tf32_on_ampere>`.
+
+    See :class:`~torch.nn.Conv2d` for details and output shape.
+
+    """
+    px, py = padding
```

Comment:
padding is defined as just an int in the signature but used as a tuple?

Reply:
hmm. I just moved  this code as is so will leave changes to future prs.